manyfews/calculations/tasks.py

```diff
@@ -81,26 +82,68 @@ def runningGenerateRiverFlows(dataDate, dataLocation):
         dataFileDirPath, "RainfallRunoffModelParameters.csv"
     )
 
-    gefsData = prepareGEFSdata(date=dataDate, location=dataLocation)
-    initialConditionData = prepareInitialCondition(date=dataDate, location=dataLocation)
+    # plus time zone information
+    beginDate = datetime.astimezone(beginDate, tz=timezone(timedelta(hours=0)))
+
+    # prepare GEFS data for model.
+    gefsData = prepareGEFSdata(date=beginDate, location=dataLocation)
+
+    # prepare initial condition data for model.
+    initialConditionData = prepareInitialCondition(
+        date=beginDate, location=dataLocation
+    )
+
+    # run model.
     riverFlowsData = GenerateRiverFlows(
+        dt=0.25,
         gefsData=gefsData,
         F0=initialConditionData,
         parametersFilePath=parametersFilePath,
     )
 
-    # import the next day's initial condition data into DB.
-    # save into DB ( 'calculations_initialcondition' table)
+    # riverFlowsData[0] ====> Q: River flow (m3/s).
+    # riverFlowsData[1] ====> qp: Rainfall (mm/day).
+    # riverFlowsData[2] ====> Ep: Potential evapotranspiration (mm/day).
+    # riverFlowsData[3] ====> F0: intial condition data for next day.
+
+    riverFlows = riverFlowsData[0]
+    qp = riverFlowsData[1]
+    Ep = riverFlowsData[2]
     F0 = riverFlowsData[3]  # next day's initial condition
-    tomorrow = dataDate + timedelta(days=1)
 
+    # import the next day's initial condition data F0 into DB.
+    # ('calculations_initialcondition' table)
+    nextDay = beginDate + timedelta(days=1)
     for i in range(len(F0[:, 0])):
         nextDayInitialCondition = InitialCondition(
-            date=tomorrow,
+            date=nextDay,
             location=dataLocation,
             storage_level=F0[i, 0],
             slow_flow_rate=F0[i, 1],
             fast_flow_rate=F0[i, 2],
         )
         nextDayInitialCondition.save()
-    return riverFlowsData
+
+    for i in range(qp.shape[0]):
+        # save qp and Eq and into DB.
+        # ( 'calculations_rainandevapotranspiration' table)
+        forecastHour = int(i * 24 * dt)
+        RainAndEvapotranspirationData = RainAndEvapotranspiration(
+            date=beginDate,
+            location=dataLocation,
+            rain_fall=qp[i],
+            potential_evapotranspiration=Ep[i],
+            forecast_hour=forecastHour,
+        )
+        RainAndEvapotranspirationData.save()
+
+        # save Q into DB.
+        # ('calculations_potentialriverflows' table)
+        for j in range(riverFlows.shape[1]):
+            PotentialRiverFlowsData = PotentialRiverFlows(
+                date=beginDate,
+                location=dataLocation,
+                river_flows=riverFlows[i, j],
+                forecast_hour=forecastHour,
+            )
```

Comment:
It looks like we are not saving the index of each individual prediction, so currently I don't think we could retrieve a single prediction time series from the database. I think we should add in a `prediction_index` or similar?